src/DotLiquid/Strainer.cs

```diff
@@ -15,6 +15,27 @@ static class DictionaryExtensions
                 return dic[key] = factory();
             return found;
         }
+
+    }
+
+    static class MethodInfoExtensions
+    {
+        public static int GetNonContextParameterCount(this MethodInfo method)
+        {
+            return method.GetParameters().Count(p => p.ParameterType != typeof(Context));
+        }
+
+        public static bool MatchesMethod(this MethodInfo method, KeyValuePair<string, IList<Tuple<object, MethodInfo>>> compareMethod)
+        {
+            string methodName = Template.NamingConvention.GetMemberName(method.Name);
+            if (compareMethod.Key != methodName)
+            {
+                return false;
+            }
+            int methodParamCount = method.GetNonContextParameterCount();
+
+            return compareMethod.Value.Any(m => m.Item2.GetNonContextParameterCount() == methodParamCount);

```

Comment:
should you also be checking if the parameter types are matching, not just their count?

Reply:
@daviburg We historically didn't do type matching. Liquid doesn't really care and most of the time we accept object. It would be a good enhancement as we can optimize lots of code that does casting unnecessarily, but it wouldn't be in scope for this.

Reply:
fair enough